### PR TITLE
[MCP] Use MCRegUnit as the key type of CopyTracker::Copies map. NFC.

### PR DESCRIPTION
**Pull request by bzEq**
`CopyTracker` is in fact tracking at RegUnit level, not MCRegister.

### CODE REVIEW

**Review by spaits — APPROVED** [review]
Even though it has worked with `MCRegister` it is better to have the correct type. The changes look good to me. I don't know if my approval is enough or we should also wait for an approval from @qcolombet .

**Comment by bzEq** [comment]
Thanks for quick review, I'll leave it open for a couple of days.

**Review by arsenm — APPROVED** [review]

**Review by qcolombet — APPROVED** [review]

**Pull request merged**
